src/background/utils/preinject.js

```diff
@@ -282,9 +301,17 @@ function prepareScript(script) {
   const reqsSlices = reqs ? [].concat(...reqs.map(req => [req, '\n;'])) : [];
   const hasReqs = reqsSlices.length;
   const wrap = !meta.unwrap;
+  const { grant } = meta;
+  const grantNone = grant.length === 1 && grant[0] === 'none';
   const injectedCode = [
+    wrap && `window.${dataKey}=function(scope,${dataKey}){
+      try{
+        with(scope)
+        ${grantNone ? '' : `with(${dataKey})`}
+        ((define,module,exports)=>{
+          ${grantNone ? '' : `delete(${dataKey}); var{${INLINED_GLOBALS}}=this;`}
```

Comment:
I dont understand this. Wouldnt this set `var Math = this`???
how do you make `window.Math` reflect this var?
also how do you set this var if `window.Math = polyfill` is executed?